### PR TITLE
Fixes infinite rods created from reinforced floors

### LINKED ISSUE #35280: Removing a reinforced floor duplicates rods infinitely
Step 1: make reinforced floor
Step 2: crowbar it up. The reinforced floor is not deleted but it creates rods
Step 3: infinite metal.

Roundid: 83030

### PR DESCRIPTION
**Pull request by Robustin**
:cl: Robustin
fix: Crow-barring a reinforced floor will no longer create (infinite) rods
/:cl:

Fixes #35280

I'm still not sure HOW this fixes it since the previous attackby didn't call the parent (which allowed rod creation), but during testing moving the floor removal to wrench_act and making the attackby return brought back the intended behavior.

### CODE REVIEW

**Comment by nicbn** [comment]
That doesn't fix my PR

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
Replace your attackby addition with crowbar_act returning and see if that works (it should)

**Comment by Robustin** [comment]
@ExcessiveUseOfCobblestone 

It didn't work, still dupes when:

>>/turf/open/floor/engine/crowbar_act(mob/user, obj/item/tool)
>>	       return

**Comment by ExcessiveUseOfCobblestone** [comment]
it must be calling inher before the tool_acts

**Comment by ExcessiveUseOfCobblestone** [comment]
Yeah this will make it unable to be hit by bibles

**Comment by nicbn** [comment]
This fixes #35280 not #35288

File: code/game/turfs/simulated/floor/reinf_floor.dm [hunk 1]
```diff
@@ -25,18 +25,21 @@
 		..()
 	return //unplateable
 
-/turf/open/floor/engine/attackby(obj/item/C, mob/user, params)
-	if(!C || !user)
+/turf/open/floor/engine/wrench_act(mob/living/user, obj/item/wrench/W)
```
Comment (ShizCalev) [inline code/game/turfs/simulated/floor/reinf_floor.dm]: iirc part of the reason for switching to wrench_act was so you wouldn't have to typecast solely to obj/item/wrench, that way you could create multitools with different paths that can call wrench_act too

usesound & toolspeed are obj/item level vars, so you should be fine to drop it down a level.

File: code/game/turfs/simulated/floor/reinf_floor.dm [hunk 2]
```diff
@@ -25,18 +25,21 @@
...
+	if(!W || !user)
```
Comment (ShizCalev) [inline code/game/turfs/simulated/floor/reinf_floor.dm]: Is this check needed? None of the other tool acts that currently exist check for these, despite doing very similar things in most cases

Comment (Robustin) [reply code/game/turfs/simulated/floor/reinf_floor.dm]: Probably not

File: code/game/turfs/simulated/floor/reinf_floor.dm [hunk 3]
```diff
@@ -34,6 +34,9 @@
 		ChangeTurf(/turf/open/floor/plating)
 	return TRUE
 
+/turf/open/floor/engine/attackby(obj/item/C, mob/user, params)
```
Comment (AnturK) [inline code/game/turfs/simulated/floor/reinf_floor.dm]: This is not a proper fix, move force param in make_plating up to /floor and pass 1 in manual removal.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/game/turfs/simulated/floor/reinf_floor.dm (hunk 3 above)
Comment (Robustin) [reply code/game/turfs/simulated/floor/reinf_floor.dm]: @AnturK 

But we don't want people to be able to crowbar reinforced flooring either.

Comment (ShizCalev) [reply code/game/turfs/simulated/floor/reinf_floor.dm]: return on crowbar_act then?

Comment (ShizCalev) [reply code/game/turfs/simulated/floor/reinf_floor.dm]: Oh, floor doesn't use crowbar_act and calls prytile directly if you attack it with a crowbar, hm.

**Comment by Supermichael777** [comment]
Parent callback is implicit at end of procs in byond. You can move it with ..() or prevent it with a return statment.

**Comment by Robustin** [comment]
@ShizCalev is doing is the "proper" way so meh

**Pull request closed without merging**